pkg/streamingpromql/operators/aggregations/aggregation.go

```diff
@@ -113,6 +118,15 @@ func (a *Aggregation) ExpressionPosition() posrange.PositionRange {
 }
 
 func (a *Aggregation) SeriesMetadata(ctx context.Context) ([]types.SeriesMetadata, error) {
+	// Pre-process any Scalar params
+	var err error
```

Comment:
[nit] We could move this declaration of `err` inside the `if a.Param != nil` block